src/jit/lower.cpp

```diff
@@ -2835,6 +2835,13 @@ void Lowering::InsertPInvokeMethodProlog()
 
     JITDUMP("======= Inserting PInvoke method prolog\n");
 
+    // If the entry block has more than one incoming edge, insert a preceding scratch BB into which we can safely
+    // insert the P/Invoke method prolog.
+    if (comp->fgFirstBB->countOfInEdges() > 1)
```

Comment:
In fact we should probably call it much earlier, as soon as we realize we're going to have inline pinvokes, and just assert that was called here.